File: packages/customer-portal/backend/src/common/entities/base.entity.ts
import {
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  UpdateDateColumn,
} from 'typeorm';

export abstract class BaseEntity {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ name: 'created_by_user_id', nullable: true, type: 'uuid' })
  createdByUserId?: string;

  @Column({ name: 'created_by_user_name', nullable: true, length: 255 })
  createdByUserName?: string;

  @CreateDateColumn({
    name: 'created_at_utc',
    type: 'timestamp with time zone',
  })
  createdAtUtc: Date;

  @Column({ name: 'modified_by_user_id', nullable: true, type: 'uuid' })
  modifiedByUserId?: string;

  @Column({ name: 'modified_by_user_name', nullable: true, length: 255 })
  modifiedByUserName?: string;

  @UpdateDateColumn({
    name: 'modified_at_utc',
    type: 'timestamp with time zone',
  })
  modifiedAtUtc: Date;
}

File: src/Shared/Shared.Kernel/BaseEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Shared.Kernel;

public abstract class BaseEntity
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }

    public Guid? CreatedByUserId { get; set; }
    
    [MaxLength(255)]
    public string? CreatedByUserName { get; set; }
    
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;

    public Guid? ModifiedByUserId { get; set; }
    
    [MaxLength(255)]
    public string? ModifiedByUserName { get; set; }
    
    public DateTime ModifiedAtUtc { get; set; } = DateTime.UtcNow;
}